src/consumers.ts

```diff
@@ -161,7 +174,50 @@ class SecurityEngine extends AnalysisConsumer implements DiagnosticProducer {
         this.highestSeverityBinding = { path: ['highest_severity'] };
     }
 
-    produce(ctx: any): Diagnostic[] {
+    private getAggregaterExploitCount(oldExploitCount: string): string {
+        // Compute sum of exploits, taking 'unavailable' value for exploit.
+        var newExploitCount: any
+        if (oldExploitCount == "unavailable") {
+            newExploitCount = this.exploitCount || "unavailable";
+        } else {
+            newExploitCount = parseInt(oldExploitCount) + this.exploitCount
+        }
+
+        return newExploitCount.toString()
+    }
+
+    private getAggregatedSeverity(oldSeverity: string): string {
+        // Compute highest severity
+        var newSeverity = oldSeverity
+        if (this.highestSeverity == "critical" || oldSeverity == "critical")
+            newSeverity = "critical"
+        else if (this.highestSeverity == "high" || oldSeverity == "high")
+            newSeverity = "high"
+        else if (this.highestSeverity == "medium" || oldSeverity == "medium")
+            newSeverity = "medium"
+        else if (this.highestSeverity == "low" || oldSeverity == "low")
+            newSeverity = "low"
+
+        return newSeverity
+    }
+
+    private getMaxRecVersion(oldRecVersion: string): string {
+        var newRecVersion = oldRecVersion;
+
+        // Compute rec version based on max criteria
+        var newRecVersion = oldRecVersion
+        if (oldRecVersion == "N/A") {
+            newRecVersion = this.changeTo || "N/A";
+        } else if (this.changeTo) {
+            if (compareVersions(oldRecVersion, this.changeTo) == -1) {
+                newRecVersion = this.changeTo
+            }
+        }
```

Comment:
Bit confused here. Is the Recommendation at module level or package level?

Reply:
Recommendation is always at module level, that's what we can change in go.mod file. Here we are computing max recommendation version for module and all its packages together.

Reply:
I was concerned about the recommendation from Graph. Is it at module level?

Reply:
Yes, from Graph its always at Module level.